Fixes & improvements to Subtitle Chooser

svobs opened the pull request:
- [x] I have read [CONTRIBUTING.md](https://github.com/iina/iina/blob/develop/CONTRIBUTING.md)
- [x] This implements/fixes issue #4057.

---

**Description:**
Like #4825 but makes additional changes:
- The message `osd.sub_found`, previously set to `%d subtitles found. Downloading…`, was not exactly appropriate as the heading for the list of subtitles to download. It looks like this was done because it was used both for the subtitle list & for the actual "downloading" message. So I changed the existing `osd.sub_found` message to remove the ` Downloading…` part, added a new `osd.sub_downloading` message whose English version is `Downloading %d subtitles`, and changed the "downloading" message to use this key instead.
- Improve text size of sub-heading and table items slightly.
- Change table style & border so it looks like [this mock-up](https://github.com/iina/iina/issues/4057#issuecomment-1949629135) from @Iiii-I-I-I. Also implemented some other changes to displayed attributes based on said mock-up:
- Remove time from upload date (seems rare that subtitles are uploaded more than once a day).
- Remove rating (seems to be 0.0 most of the time?) - am open to bringing this back.
- Add release year in parentheses.
- Add up to 2 fraction digits to `fps` attribute's displayed value.

svobs commented:
Squashed + rebased to fix merge errors + improved commit msg.

lhc70000 commented on iina/Base.lproj/Localizable.strings:
```diff
@@ -83,7 +83,8 @@
 "osd.sub_not_found" = "No subtitles found";
-"osd.sub_found" = "%d subtitles found. Downloading…";
+"osd.sub_found" = "%d subtitles found.";
+"osd.sub_downloading" = "Downloading %d subtitles";
```
Please also add this to en.lproj.

lhc70000 commented on iina/Extensions.swift:
```diff
@@ -290,6 +299,12 @@ extension FloatingPoint {
   }
+
+  /// Formats as String, rounding the number to 2 digits after the decimal
+  var stringMaxFrac2: String {
```
I suggest using a name like `stringWithMaxFractionDigit2` for better readability.

svobs replied:
Ah yeah. For utility functions like this which might be commonly used, I worry about them getting too long. But, just semantics. Done

lhc70000 commented on iina/OSDMessage.swift:
```diff
@@ -69,6 +69,7 @@ enum OSDMessage {
 
   case startFindingSub(String)  // sub source
   case foundSub(Int)
+  case downloadingSub(Int, String)  // download count, ssub source
```
sub

svobs replied:
Thanks

lhc70000 committed:
Fix conflicting constraints in subtitle sidebar, #5027

svobs commented:
Rebased + pushed an update with all the suggested changes and more detailed commit message.

svobs committed:
Enhancements to online subtitle download OSD

  Details:
  • Improve text size of sub-heading and table items slightly
  • Improve table style & border
  • Remove time from upload date
  • Remove rating from displayed attributes
  • Add release year in parentheses
  • Add up to 2 fraction digits to fps attribute's displayed value
  • Add osd.sub_downloading ("Downloading %d subtitles") & reference it
    while downloading instead of osd.sub_found
  • Remove " Downloading…" from osd.sub_found to make it more
    appropriate as a table heading

lhc70000 commented:
Merged via 1f87c39.

low-batt commented:
This was included in IINA 1.4.0.
